marlowe-playground-client/src/MainFrame/State.purs

```diff
@@ -228,9 +242,21 @@ handleQuery ::
   Query a ->
   HalogenM State Action ChildSlots Void m (Maybe a)
 handleQuery settings (ChangeRoute route next) = do
-  handleRoute settings route
+  -- Without the following each route is handled twice, once when we call selectView ourselves
+  -- and another which is triggered in Main, when the route changes.
+  currentView <- use _view
+  when (routeToView route /= Just currentView) $ handleRoute settings route
   pure $ Just next
 
+-- FIXME: Delete before merging
+-- showHandleAction ::
+--   forall m.
+--   MonadAff m =>
+--   SPSettings_ SPParams_ ->
+--   Action ->
+--   HalogenM State Action ChildSlots Void m Unit
+-- showHandleAction settings action = handleAction settings $ spy "MainFrame.handleAction" action
+-- TODO: Should we refactor the settings so that it's a ReaderT or part of the State?
```

Comment:
The first option requires we use our own monad rather than `HalogenM` (which is still an option), the second is an idea, it might be nicer in some places.